### PR TITLE
PFM-1099: Added swagger doc of muster roll service

### PR DESCRIPTION
**Pull request by rohit-eGov**

### CODE REVIEW

**Review by rushang7-eGov — CHANGES_REQUESTED** [review]
- Delete the .DS_STORE file
- Execute **Reformat Code**
- Add .gitignore so that all the build target classes are removed from git.

File: backend/muster-roll-service/pom.xml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    <version>1.0.0</version>
+    <properties>
+    <java.version>1.8</java.version>
+    <maven.compiler.source>${java.version}</maven.compiler.source>
+    <maven.compiler.target>${java.version}</maven.compiler.target>
+</properties>
+    <parent>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.2.6.RELEASE</version>
```
[the remark this answers is not in the document]
Comment (rohit-eGov) [reply backend/muster-roll-service/pom.xml]: Done

File: backend/muster-roll-service/src/main/java/org/egov/kafka/Consumer.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.support.KafkaHeaders;
+import org.springframework.messaging.handler.annotation.Header;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+
+@Component
+public class Consumer {
```
[the remark this answers is not in the document]
Comment (rohit-eGov) [reply backend/muster-roll-service/src/main/java/org/egov/kafka/Consumer.java]: Removed

File: backend/muster-roll-service/src/main/java/org/egov/util/IdgenUtil.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
+package org.egov.util;
```
[the remark this answers is not in the document]
Comment (rohit-eGov) [reply backend/muster-roll-service/src/main/java/org/egov/util/IdgenUtil.java:1]: Removed unnecessary util class

File: backend/muster-roll-service/src/main/java/org/egov/web/controllers/MusterRollApiController.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import java.io.IOException;
+
+import javax.validation.constraints.*;
+    import javax.validation.Valid;
+    import javax.servlet.http.HttpServletRequest;
+
+@javax.annotation.Generated(value = "org.egov.codegen.SpringBootCodegen", date = "2022-11-14T19:58:09.415+05:30")
+
+@Controller
+    @RequestMapping("/muster-roll/v1")
```
[the remark this answers is not in the document]
Comment (rohit-eGov) [reply backend/muster-roll-service/src/main/java/org/egov/web/controllers/MusterRollApiController.java]: Updated

File: backend/muster-roll-service/src/main/java/org/egov/web/controllers/MusterRollApiController.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
...etadata in the request body"  )  @Valid @RequestBody RequestInfo requestInfo,@ApiParam(value = "Id of the muster roll") @Valid @RequestParam(value = "id", required = false) String id,@ApiParam(value = "(Custom-formatted) Muster roll number") @Valid @RequestParam(value = "musterRollNumber", required = false) String musterRollNumber,@ApiParam(value = "Register Id") @Valid @RequestParam(value = "registerId", required = false) String registerId,@ApiParam(value = "Return registers with any overlap in the given time period") @Valid @RequestParam(value = "fromDate", required = false) BigDecimal fromDate,@ApiParam(value = "Return registers with any overlap in the given time period") @Valid @RequestParam(value = "toDate", required = false) BigDecimal toDate,@ApiParam(value = "Status of the muster roll. This can't be the only query param. It should be paired with some other search param.") @Valid @RequestParam(value = "status", required = false) String status,@ApiParam(value = "Workflow Status of the muster roll. This can't be the only query param. It should be paired with some other search param.") @Valid @RequestParam(value = "musterRollStatus", required = false) String musterRollStatus) {
```
[the remark this answers is not in the document]
Comment (rohit-eGov) [reply backend/muster-roll-service/src/main/java/org/egov/web/controllers/MusterRollApiController.java]: Updated

File: backend/muster-roll-service/src/main/java/org/egov/web/models/AttendanceEntry.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+@javax.annotation.Generated(value = "org.egov.codegen.SpringBootCodegen", date = "2022-11-14T19:58:09.415+05:30")
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class AttendanceEntry   {
+        @JsonProperty("time")
+        private BigDecimal time = null;
```
[the remark this answers is not in the document]
Comment (rohit-eGov) [reply backend/muster-roll-service/src/main/java/org/egov/web/models/AttendanceEntry.java]: Updated

File: backend/muster-roll-service/src/main/resources/application.properties [hunk 7]
```diff
@@ -0,0 +1,88 @@
+server.contextPath=/muster-roll-service
```
[the remark this answers is not in the document]
Comment (rohit-eGov) [reply backend/muster-roll-service/src/main/resources/application.properties]: Updated

File: backend/muster-roll-service/src/main/resources/application.properties [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+kafka.consumer.config.auto_commit_interval=100
+kafka.consumer.config.session_timeout=15000
+kafka.consumer.config.auto_offset_reset=earliest
+#-----------------KAFKA PRODUCER CONFIGURATIONS--------------------------------#
+kafka.producer.config.retries_config=0
+kafka.producer.config.batch_size_config=16384
+kafka.producer.config.linger_ms_config=1
+kafka.producer.config.buffer_memory_config=33554432
+
+#Localization config
```
[the remark this answers is not in the document]
Comment (rohit-eGov) [reply backend/muster-roll-service/src/main/resources/application.properties:45]: Not removed much, but as development progress will remove some unnecessary configs

File: backend/muster-roll-service/src/main/java/org/egov/web/models/MusterRollRequestWorkflow.java [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+ */
+@Validated
+@javax.annotation.Generated(value = "org.egov.codegen.SpringBootCodegen", date = "2022-11-14T19:58:09.415+05:30")
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class MusterRollRequestWorkflow   {
```
[the remark this answers is not in the document]
Comment (rohit-eGov) [reply backend/muster-roll-service/src/main/java/org/egov/web/models/MusterRollRequestWorkflow.java]: Updated

File: backend/muster-roll-service/pom.xml [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+            <groupId>com.fasterxml.jackson.datatype</groupId>
+            <artifactId>jackson-datatype-jsr310</artifactId>
+        </dependency>
+    <!-- Bean Validation API support -->
+        <dependency>
+            <groupId>javax.validation</groupId>
+            <artifactId>validation-api</artifactId>
+        </dependency>
+    </dependencies>
+    <repositories>
```
[the remark this answers is not in the document]
Comment (rohit-eGov) [reply backend/muster-roll-service/pom.xml:101]: Replaced

File: backend/muster-roll-service/src/main/java/org/egov/web/controllers/MusterRollApiController.java [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+import java.io.IOException;
+
+import javax.validation.constraints.*;
+    import javax.validation.Valid;
+    import javax.servlet.http.HttpServletRequest;
+
+@javax.annotation.Generated(value = "org.egov.codegen.SpringBootCodegen", date = "2022-11-14T19:58:09.415+05:30")
+
+@Controller
+    @RequestMapping("/v1")
```
Comment (rushang7-eGov) [inline backend/muster-roll-service/src/main/java/org/egov/web/controllers/MusterRollApiController.java:28]: Please run Reformat Code on all files to adjust tab spaces.

**Review by rushang7-eGov — CHANGES_REQUESTED** [review]

File: backend/muster-roll-service/src/main/java/org/egov/web/controllers/MusterRollApiController.java (hunk 11 above)
Comment (rohit-eGov) [reply backend/muster-roll-service/src/main/java/org/egov/web/controllers/MusterRollApiController.java]: Applied refactor code on all files

**Review by rushang7-eGov — APPROVED** [review]

**Pull request merged**